tools/azure-sdk-tools/devtools_testutils/azure_recorded_testcase.py

```diff
@@ -271,11 +271,11 @@ def get_credential(**kwargs):
                 **kwargs
             )
         # This is for testing purposes only, to ensure that the AzurePipelinesCredential is used when available
-        # else:
-        #     raise ValueError(
-        #         "Environment variables not set for service principal authentication. "
-        #         f"service_connection_id: {service_connection_id}, client_id: {client_id}, tenant_id: {tenant_id}, system_access_token: {system_access_token}"
-        #     )
+        else:
+            raise ValueError(
+                "Environment variables not set for service principal authentication. "
+                f"service_connection_id: {service_connection_id}, client_id: {client_id}, tenant_id: {tenant_id}, system_access_token: {system_access_token}"
```

Comment:
We should not be dumping the system access token here.

Reply:
This is for testing purposes (as in the comments) to help us understand why APC does not work.

Reply:
We enable these just for temporary investigation. After we have done with the investigation, we will comment these lines to stop dumping the system access token.

Reply:
Even in that case we should only dump if the token exists and not the value.

Reply:
Agreed. Real values have no value. Existence is the information we want to know.

Reply:
https://github.com/Azure/azure-sdk-for-python/pull/36528, sanitize the token before putting it in log.